// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

package net;

option optimize_for = LITE_RUNTIME;

import "public.proto";

// A TrustTokenIssuerConfig message represents persistent state scoped
// to a single Trust Tokens issuer origin.
// An issuer’s config contains:
// - any signed-but-unspent tokens received from the issuer; and
// - the issuer’s “batch size” (number of blinded tokens to send
// per issuance request), if it has configured one via its key
// commitment endpoint.
message TrustTokenIssuerConfig {
  // Keys the issuer has recently committed (no need to store stale keys)
  repeated TrustTokenKeyCommitment keys = 1;
  optional int32 batch_size = 2;
  repeated TrustToken tokens = 3;
  // The time of the most recent issuance for this pair. Used for
  // rate-limiting.
  optional string last_issuance = 4;
}

// A top-level domain’s config currently just stores
// the issuers with which the domain has associated itself.
message TrustTokenToplevelConfig {
  repeated string associated_issuers = 1;
}

// An TrustTokenIssuerToplevelPairConfig stores state scoped to
// (issuer origin, top-level page origin) pairs.
message TrustTokenIssuerToplevelPairConfig {
  // The time of the most recent redemption for this pair. Used for
  // rate-limiting.
  optional string last_redemption = 1;
  // The signed redemption record cached for this pair, if any.
  optional SignedTrustTokenRedemptionRecord signed_redemption_record = 2;
}
